vowpalwabbit/rand48.cc

```diff
@@ -26,3 +27,18 @@ float merand48(uint64_t& initial)
 }
 
 float merand48_noadvance(uint64_t v) { return merand48(v); }
+
+float merand48_boxmuller(uint64_t& index)
+{
+  static float x1 = 0.0;
```

Comment:
These being null makes this non-threadsafe. They are local state required for this right? I'm wondering if there is another way to do this to avoid that constraint

Reply:
They don't have to be static.  I copied this from another file into here to reuse the code, and I didn't notice that.